### PR TITLE
2.1.0

### PR DESCRIPTION
**Pull request by savkk**

### CODE REVIEW

File: qase-testng/src/test/java/io/qase/testng/QaseListenerTest.java [hunk 1]
```diff
@@ -57,6 +57,23 @@ public void passedBulkTest() {
...
     }
 
+    @Test
+    public void passedWithStepsBulkTest() {
+        useBulk(true);
+        runTest(WithSteps.class);
+        verify(postRequestedFor(urlPathEqualTo("/v1/result/PRJ/777/bulk"))
+                .withHeader("Token", equalTo("secret-token"))
+                .withHeader("Content-Type", equalTo("application/json; charset=UTF-8"))
+                .withRequestBody(equalToJson("{\n" +
```
Comment (ilyafilinin) [inline qase-testng/src/test/java/io/qase/testng/QaseListenerTest.java:67]: I would suggest writing Json into a separate object and then comparing it with an object, not a huge json

File: qase-testng/src/test/java/io/qase/testng/QaseListenerTest.java [hunk 2]
```diff
@@ -115,7 +132,8 @@ public void failedWithTimeBulkTest() {
     @Test
     public void allBulkTests() {
         useBulk(true);
-        runTest(Arrays.asList(Passed.class, PassedWithTime.class, Failed.class, FailedWithTime.class));
+        runTest(Arrays.asList(Passed.class, PassedWithTime.class, Failed.class, FailedWithTime.class,
+                WithSteps.class, WithStepsSuccess.class));
         verify(postRequestedFor(urlPathEqualTo("/v1/result/PRJ/777/bulk"))
                 .withHeader("Token", equalTo("secret-token"))
                 .withHeader("Content-Type", equalTo("application/json; charset=UTF-8"))
```
Comment (ilyafilinin) [inline qase-testng/src/test/java/io/qase/testng/QaseListenerTest.java:139]: I would suggest writing Json into a separate object and then comparing it with an object, not a huge json

File: qase-testng/src/test/java/io/qase/testng/QaseListenerTest.java [hunk 3]
```diff
@@ -57,6 +57,23 @@ public void passedBulkTest() {
                         "}")));
     }
 
+    @Test
+    public void passedWithStepsBulkTest() {
+        useBulk(true);
+        runTest(WithSteps.class);
+        verify(postRequestedFor(urlPathEqualTo("/v1/result/PRJ/777/bulk"))
```
[the remark this answers is not in the document]
Comment (savkk) [reply qase-testng/src/test/java/io/qase/testng/QaseListenerTest.java]: I don't know how to do it in wiremock, only if I will marshal object through gson or jackson and after it deserialize object to json. But I think it isn't a good idea.
Can you text how you see it?

File: qase-testng/src/test/java/io/qase/testng/QaseListenerTest.java (hunk 1 above)
Comment (savkk) [reply qase-testng/src/test/java/io/qase/testng/QaseListenerTest.java:69]: I don't know how to do it in wiremock, only if I will marshal object through gson or jackson and after it deserialize object to json. But I think it isn't a good idea.
Can you text how you see it?

Comment (ilyafilinin) [reply qase-testng/src/test/java/io/qase/testng/QaseListenerTest.java:68]: I was moving to another city, I could not answer earlier, sorry.
Previously, it seems, I did not use WireMock, now I looked at the docks on it and also did not immediately find how to do it normally. I assumed that you can "shove" json into an object and compare through the equals method with it.
Apparently not.

**Pull request merged**